multimodal/src/autogluon/multimodal/learners/object_detection.py

```diff
@@ -0,0 +1,733 @@
+
+import pandas as pd
+from omegaconf import OmegaConf
+from torch import nn
+
+from ..constants import (
+    BBOX,
+    DDP,
+    MAP,
+    MULTI_IMAGE_MIX_DATASET,
+    OBJECT_DETECTION,
+    OPEN_VOCABULARY_OBJECT_DETECTION,
+    OVD_RET,
+    XYWH,
+)
+from ..data import BaseDataModule, MultiImageMixDataset, MultiModalFeaturePreprocessor, infer_rois_column_type
+from ..optimization import MMDetLitModule
+from ..utils import (
+    check_if_packages_installed,
+    cocoeval,
+    compute_inference_batch_size,
+    compute_num_gpus,
+    convert_pred_to_xywh,
+    create_fusion_model,
+    extract_from_output,
+    from_coco_or_voc,
+    get_detection_classes,
+    infer_precision,
+    object_detection_data_to_df,
+    save_ovd_result_df,
+    save_result_df,
+    setup_save_path,
+    split_train_tuning_data,
+)
+from .base import BaseLearner
+
+logger = logging.getLogger(__name__)
+
+
+class ObjectDetectionLearner(BaseLearner):
+    def __init__(
+        self,
+        label: Optional[str] = None,
+        problem_type: Optional[str] = OBJECT_DETECTION,
+        presets: Optional[str] = None,
+        eval_metric: Optional[str] = None,
+        hyperparameters: Optional[dict] = None,
+        path: Optional[str] = None,
+        verbosity: Optional[int] = 2,
+        num_classes: Optional[int] = None,  # TODO: can we infer this from data?
+        classes: Optional[list] = None,
+        warn_if_exist: Optional[bool] = True,
+        enable_progress_bar: Optional[bool] = None,
+        pretrained: Optional[bool] = True,
+        sample_data_path: Optional[str] = None,
+    ):
+        super().__init__(
+            problem_type=problem_type,
+            presets=presets,
+            eval_metric=eval_metric,
+            hyperparameters=hyperparameters,
+            path=path,
+            verbosity=verbosity,
+            num_classes=num_classes,
+            classes=classes,
+            warn_if_exist=warn_if_exist,
+            enable_progress_bar=enable_progress_bar,
+            pretrained=pretrained,
+            sample_data_path=sample_data_path,
+        )
+        check_if_packages_installed(problem_type=self._problem_type)
+
+        # TODO: merge object detection and open vocabulary object detection
+        if self._problem_type == OBJECT_DETECTION:
+            self._label_column = "label"
+            if self._sample_data_path is not None:
+                self._classes = get_detection_classes(self._sample_data_path)
+                self._output_shape = len(self._classes)
+
+        # TODO: merge _detection_anno_train and detection_anno_train?
+        self._detection_anno_train = None
+        self.detection_anno_train = None
+
+    @property
+    def classes(self):
+        """
+        Return the classes of object detection.
+        """
+        return self._model.model.CLASSES
+
+    def setup_detection_train_tuning_data(self, max_num_tuning_data, seed, train_data, tuning_data):
```

Comment:
Need a refactor for this function. I'll do it in a PR later.

Reply:
Yeah, it becomes easier to check the logic and optimize it.